### PR TITLE
chore: support creating qemu guest socket

### PR DESCRIPTION
**Pull request by frezbo**
Support creating a qemu guest agent socket so we can test `qemu-guest-agent` extension in CI.

Ref: https://github.com/siderolabs/extensions/pull/173#issuecomment-1611911106

### CODE REVIEW

File: pkg/provision/providers/qemu/launch.go [hunk 1]
```diff
@@ -257,6 +258,18 @@ func launchVM(config *LaunchConfig) error {
 		"-smbios", fmt.Sprintf("type=1,uuid=%s", config.NodeUUID),
 	}
 
+	if config.QEMUGuestSocketEnabled {
+		args = append(
```
Comment (smira) [inline pkg/provision/providers/qemu/launch.go:262]: should we just always create the guest agent socket? is there any harm in enabling it?

Comment (frezbo) [reply pkg/provision/providers/qemu/launch.go:262]: no, it's fine, I'll update then

Comment (smira) [reply pkg/provision/providers/qemu/launch.go:262]: I mean we could simplify by having it enabled always, one less flag to support :-D

**Review by smira — APPROVED** [review]

**Pull request merged**